src/Libraries/Microsoft.Extensions.Http.Diagnostics/Logging/Internal/Log.cs

```diff
@@ -46,17 +46,114 @@ public static void OutgoingRequestError(ILogger logger, LogRecord record, Except
         OutgoingRequest(logger, LogLevel.Error, 2, nameof(OutgoingRequestError), record, exception);
     }
 
-    [LoggerMessage(LogLevel.Error, RequestReadErrorMessage)]
-    public static partial void RequestReadError(ILogger logger, Exception exception, HttpMethod httpMethod, string? httpHost, string? httpPath);
+    public static void RequestReadError(ILogger logger, Exception exception, HttpMethod method, string? host, string? path)
+    {
+        var state = LoggerMessageHelper.ThreadLocalState;
+
+        _ = state.ReserveTagSpace(4);
+        state.TagArray[3] = new(HttpClientLoggingTagNames.Method, method);
+        state.TagArray[2] = new(HttpClientLoggingTagNames.Host, host);
+        state.TagArray[1] = new(HttpClientLoggingTagNames.Path, path);
+        state.TagArray[0] = new(OriginalFormat, RequestReadErrorMessage);
+
+        logger.Log(
+            LogLevel.Error,
+            new(0, nameof(RequestReadError)),
+            state,
+            exception,
+            static (s, _) =>
+            {
+                var method = s.TagArray[3].Value ?? NullString;
+                var host = s.TagArray[2].Value ?? NullString;
+                var path = s.TagArray[1].Value ?? NullString;
+                return FormattableString.Invariant(
+                    $"An error occurred while reading the request data to fill the logger context for request: {method} {host}/{path}");
+            });
+
+        state.Clear();
+    }
 
-    [LoggerMessage(LogLevel.Error, ResponseReadErrorMessage)]
-    public static partial void ResponseReadError(ILogger logger, Exception exception, HttpMethod httpMethod, string httpHost, string httpPath);
+    public static void ResponseReadError(ILogger logger, Exception exception, HttpMethod method, string host, string path)
+    {
+        var state = LoggerMessageHelper.ThreadLocalState;
 
-    [LoggerMessage(LogLevel.Error, LoggerContextMissingMessage)]
-    public static partial void LoggerContextMissing(ILogger logger, Exception? exception, string requestState, HttpMethod httpMethod, string? httpHost);
+        _ = state.ReserveTagSpace(4);
+        state.TagArray[3] = new(HttpClientLoggingTagNames.Method, method);
+        state.TagArray[2] = new(HttpClientLoggingTagNames.Host, host);
+        state.TagArray[1] = new(HttpClientLoggingTagNames.Path, path);
+        state.TagArray[0] = new(OriginalFormat, ResponseReadErrorMessage);
```

Comment:
same here and in other places where exception is reported - `exception.*` attributes would be awesome

Reply:
This was implemented in another PR #4584.